### ISSUE TITLE
[release/2.12] performance regressions compared with 2.11

### ISSUE THREAD

**Issue by mengfei25** [intel/torch-xpu-ops#3432]
### 🐛 Describe the bug

Observed with 2.12 rc wheel on PVC, some models got performance dropped compared with 2.11

#### Higher is better
suite | data_type | mode | model | inductor_ratio | eager_ratio
-- | -- | -- | -- | -- | --
timm_models | amp_fp16 | inference | tnt_s_patch16_224 | 0.619 | 1.003
timm_models | float16 | inference | tnt_s_patch16_224 | 0.622 | 1.02
timm_models | bfloat16 | inference | tnt_s_patch16_224 | 0.636 | 1.016
timm_models | amp_bf16 | inference | tnt_s_patch16_224 | 0.651 | 1.007
torchbench | amp_fp16 | training | vgg16 | 0.736 | 0.998
torchbench | float16 | training | vgg16 | 0.736 | 0.998
torchbench | amp_bf16 | training | vgg16 | 0.764 | 1.004
torchbench | bfloat16 | training | vgg16 | 0.776 | 0.994
torchbench | float16 | inference | hf_Roberta_base | 0.817 | 0.688
torchbench | float16 | inference | phlippe_densenet | 0.823 | 1.204
torchbench | amp_fp16 | inference | phlippe_densenet | 0.824 | 1.101
torchbench | amp_fp16 | inference | hf_Roberta_base | 0.827 | 1.031
torchbench | float32 | inference | phlippe_densenet | 0.836 | 1.172
timm_models | float16 | training | gluon_inception_v3 | 0.847 | 0.997
timm_models | amp_fp16 | training | gluon_inception_v3 | 0.85 | 0.995
timm_models | amp_fp16 | training | inception_v3 | 0.851 | 0.998
timm_models | float16 | training | adv_inception_v3 | 0.857 | 0.998
timm_models | bfloat16 | training | gluon_inception_v3 | 0.858 | 1
timm_models | float16 | inference | gmlp_s16_224 | 0.869 | 1.008
timm_models | bfloat16 | training | adv_inception_v3 | 0.871 | 0.997
timm_models | bfloat16 | training | inception_v3 | 0.873 | 1
timm_models | float16 | training | inception_v3 | 0.878 | 1.002
timm_models | amp_fp16 | training | ese_vovnet19b_dw | 0.88 | 1.007
timm_models | bfloat16 | inference | gmlp_s16_224 | 0.887 | 0.94
timm_models | float16 | training | dla102 | 0.892 | 0.993
timm_models | bfloat16 | training | ese_vovnet19b_dw | 0.897 | 1.001
timm_models | bfloat16 | inference | xcit_large_24_p8_224 | 0.938 | 0.546
timm_models | float16 | training | xcit_large_24_p8_224 | 0.968 | 0.643
timm_models | float16 | inference | xcit_large_24_p8_224 | 0.97 | 0.541
timm_models | bfloat16 | training | xcit_large_24_p8_224 | 0.998 | 0.695

Reproducer
```
docker run -it -e TZ=Aisa/Shanghai --device=/dev/mem --device=/dev/dri --group-add video --privileged --shm-size=8g intelgpu/ubuntu-24.04-lts2:2523.40 bash

curl -LsSf https://astral.sh/uv/install.sh | env UV_INSTALL_DIR="/usr/local/bin" sh
uv venv myvenv --python 3.12 --clear
source myvenv/bin/activate
uv pip install pip numpy 'setuptools<81' wheel

uv pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/test/xpu  # 2.12
# pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/xpu  # 2.11

pytorch_commit="$(python -c 'import torch; print(torch.version.git_version)')"
git clone https://github.com/pytorch/pytorch
cd pytorch
git checkout ${pytorch_commit}
# modify model list for all models
rm -rf torch torch-xpu-ops
git clone https://github.com/intel/torch-xpu-ops
rsync -avz ./torch-xpu-ops/.ci/benchmarks/ ./benchmarks/dynamo/

# install requirements
uv pip install pip pandas psutil scipy requests
uv pip install -U numpy==1.26.4
uv pip install transformers==4.55.2 accelerate
uv pip install timm==1.0.19
# optional for torchbench
git clone https://github.com/pytorch/benchmark tmp-benchmark
cd tmp-benchmark
python install <model name>

python ./benchmarks/dynamo/timm_models.py --performance --bfloat16 --inference -d xpu -n 10 --only tnt_s_patch16_224 --backend=inductor --cold-start-latency --timeout 10800 --disable-cudagraphs --output output.csv

```

### Versions

pytorch 2.12 rc
pytorch 2.11

**Comment by weishi-deng** [comment]
regression of tnt_s_patch16_224: raised by the inductor SDPA kernels(bmm+softmax -> micro_sdpa). OP benchmark has been assigned to @LuFinch.

**Comment by LuFinch** [comment]
MFDNN-14987

**Comment by jianyizh** [comment]
vgg16: pytorch pr 167318 changes max_pool2d_with_indices_backward from fallback to atomic add triton op. We may not atomic

**Comment by jianyizh** [comment]
hf_Roberta_base: 1. gemm **MFDNN-14993**，2. jira **HUGGINGFAD-941** attention mask is broadcasted on last dim (kv seq_len), sdpa need stride of last dim=1.  need change transformers https://github.com/huggingface/transformers/blob/main/src/transformers/masking_utils.py#L520 attention_mask = attention_mask.expand(batch_size, -1, q_length, kv_length) 3. jira **MFDNN-14997** broadcasted mask will not trigger microsdpa kernel in onednn onednn_verbose,v1,graph,exec,gpu,100079,sdp,matmul_qk;scale_mul;mask_add;softmax;matmul_v,,in0_f16:0:strided:undef:8x12x512x64:393216s64s768s1 in1_f16:1:strided:undef:8x12x512x64:393216s64s768s1 in2_f32:2:strided:host_scalar:0:2 in3_f16:4:strided:undef:1x1x512x1:512s512s1s1 in4_f16:5:strided:undef:8x12x512x64:393216s64s768s1 out0_f16:6:strided:undef:8x12x512x64:393216s64s768s1,fpm:strict,larger_partition_kernel_t,dnnl_backend,0.39502

**Comment by chuanqi129** [comment]
Need consider acceptance methodology

**Comment by weishi-deng** [comment]
adv_inception_v3, gluon_inception_v3,inception_v3: similar with vgg16, suffer from atomic in max_pool2d_with_indices_backward, fix by Jianyi's PR

**Comment by weishi-deng** [comment]
phlippe_densenet: cannot reproduce regression on both 1100 and 1550.

**Comment by weishi-deng** [comment]
gmlp_s16_224: gemm issue in MFDNN-14993

**Comment by weishi-deng** [comment]
dla102, ese_vovnet19b_dw: maxpool backward. Can be fixed by jianyi's PR.

**Comment by weishi-deng** [comment]
xcit_large_24_p8_224: can not reproduce on both 1100 and 1550.

**Comment by orrangetabby17** [comment]
Performance drop of xcit_large_24_p8_224 might be caused by differences in AMP behavior between the benchmark versions.

1. `force_amp_for_fp16_bf16_models`  func in `benchmarks/dynamo/torchbench.py#L169-L171` 

```python
 @property
    def force_amp_for_fp16_bf16_models(self):
        return self._config["dtype"]["force_amp_for_fp16_bf16_models"]
```

2. `force_amp_for_fp16_bf16_models`  func in `benchmarks/dynamo/common.py#L1900-L1902` 

```python
    @property
    def force_amp_for_fp16_bf16_models(self):
        return set()
```

The AMP behavior is configured as follows (e.g. in release)
```python
    def cast_based_on_args(self, model, example_inputs):
        if self.args.float32 or self.args.only in self.fp32_only_models:
            if not self.args.float32:
                log.warning("Model %s supports float32 only", self.args.only)
            model, example_inputs = cast_to_fp32(model, example_inputs)
        elif self.args.float16:
            if self.args.only in self.force_amp_for_fp16_bf16_models:
                log.warning(
                    "Model %s does not support float16, running with amp instead",
                    self.args.only,
                )
                self.args.amp = True
                self.setup_amp()
            else:
                model, example_inputs = cast_to_fp16(model, example_inputs)
        elif self.args.bfloat16:
            if self.args.only in self.force_amp_for_fp16_bf16_models:
                log.warning(
                    "Model %s does not support bfloat16, running with amp instead",
                    self.args.only,
                )
                self.args.amp = True
                self.setup_amp()
            elif self.args.only in self.force_fp16_for_bf16_models:
                log.warning(
                    "Model %s does not support bfloat16, running with float16 instead",
                    self.args.only,
                )
                model, example_inputs = cast_to_fp16(model, example_inputs)
            else:
                model, example_inputs = cast_to_bf16(model, example_inputs)

        return model, example_inputs
```
As a result, in the release version, AMP does not take effect because force_amp_for_fp16_bf16_models returns an empty set. However, in the test version, AMP is enabled for `xcit_large_24_p8_224`, which may impact performance.

**Comment by orrangetabby17** [comment]
`phlippe_densenet` : identified that the test version’s inductor introduced a special Cat Kernel fusion mode, which currently generates non-fused Cat Kernels with suboptimal performance, leading to the observed regression. For example, `triton_poi_fused__native_batch_norm_legit_no_training_cat_convolution_relu_7` is replaced by `triton_poi_fused_cat_convolution_5` + `triton_poi_fused__native_batch_norm_legit_no_training_cat_relu_6`.

**Comment by orrangetabby17** [comment]
phlippe_densenet : invoke an issue in pytorch repo [issues#185346](https://github.com/pytorch/pytorch/issues/185346)
